src/lib/hooks/transactions/updater.tsx

```diff
@@ -0,0 +1,100 @@
+import { TransactionReceipt } from '@ethersproject/abstract-provider'
+import { SupportedChainId } from 'constants/chains'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import useBlockNumber, { useFastForwardBlockNumber } from 'lib/hooks/useBlockNumber'
+import { useCallback, useEffect } from 'react'
+import { retry, RetryableError, RetryOptions } from 'utils/retry'
+
+interface Transaction {
+  addedTime: number
+  receipt?: unknown
+  lastCheckedBlockNumber?: number
```

Comment:
do we want to add additional fields to match what we have in app? It gives us access to things like amounts and currencies involved (we'd only need this if we want to show advanced details about the txn anywhere else in app)

Reply:
yes, absolutely. it just isn't necessary for approval txn's. I expected you to augment the `SwapTransactionInfo` interface in future work to include those, as you worked through swap submission.

Reply:
Sure makes sense